worker/include/Utils.hpp

```diff
@@ -351,51 +351,88 @@ namespace Utils
 		}
 	};
 
+	// Simple implementation of object pool only for single objects
+	// Arrays are allocated as usual
 	template<typename T>
-	class ObjectPool
+	class ObjectPoolAllocator
 	{
+		std::shared_ptr<std::vector<T*>> pool_data;
+
 	public:
-		~ObjectPool()
+		typedef T value_type;
+		thread_local static Utils::ObjectPoolAllocator<T> Pool;
+
+		ObjectPoolAllocator()
+		{
+			pool_data = std::shared_ptr<std::vector<T*>>(
+			  new std::vector<T*>(),
+			  [](std::vector<T*>* pool)
+			  {
+				  for (auto* ptr : *pool)
+				  {
+					  std::free(ptr);
+				  }
+				  delete pool;
+			  });
+		}
+
+		template<typename U>
+		ObjectPoolAllocator(const ObjectPoolAllocator<U>& other)
+		  : pool_data(ObjectPoolAllocator<T>::Pool.pool_data)
+		{
+		}
+
+		~ObjectPoolAllocator()
 		{
-			for (auto ptr : this->pool)
-			{
-				std::free(ptr);
-			}
 		}
 
-		// Get pointer to allocated memory. This can be newly allocated memory or re-use of previously
-		// returned object. Object is not initialized and shouldn't be considered to be in a valid state.
-		T* Allocate()
+		template<typename U>
+		struct rebind
+		{
+			typedef ObjectPoolAllocator<U> other;
+		};
```

Comment:
What is this struct for?

Reply:
@nazar-pc https://en.cppreference.com/w/cpp/memory/allocator

Example: we have allocator for ```RtpPacket```, but we need allocator for ```std::shared_ptr<RtpPacket>```. This construction solve this problem.

Reply:
Looks like it was deprecated in C++17 and removed in C++20, does it still mean we need to have this in C++11 anyway?

Also does this mean our allocator will store things of variable types? I mean `RtpPacket` and `std::shared_ptr<RtpPacket>` must be of a different size and shoudn't use the same pool, right?

Reply:
@nazar-pc it's i think i try to use ```std::allocator_traits<T>::rebind_alloc<U>``` construction. It's these are equivalent constructions. Now, I'm trying to make code compatible for all `standard library``` implementations(Microsoft C++, libc++ and libstdc++).

> Also does this mean our allocator will store things of variable types? I mean RtpPacket and std::shared_ptr<RtpPacket> must be of a different size and shoudn't use the same pool, right?

It's ```template magic``` to create another allocator. In our case It's another allocator with separate memory pool.